python/ray/serve/deployment_executor_node.py

```diff
@@ -0,0 +1,78 @@
+from typing import Any, Dict, List
+
+from ray.experimental.dag import DAGNode
+from ray.serve.deployment_method_executor_node import DeploymentMethodExecutorNode
+from ray.experimental.dag.constants import DAGNODE_TYPE_KEY, PARENT_CLASS_NODE_KEY
+from ray.experimental.dag.format_utils import get_dag_node_str
+
+
+class DeploymentExecutorNode(DAGNode):
+    """The lightweight executor DAGNode of DeploymentNode that optimizes for
+    efficiency.
+
+        - We need Ray DAGNode's traversal and replacement mechanism to deal
+            with deeply nested nodes as args in the DAG
+        - Meanwhile, __init__, _copy_impl and _execute_impl are on the critical
+            pass of execution for every request.
```

Comment:
why is `_copy_impl` on the critical path for execution?

Also, as an aside these should not be underscore-prefixed if they're part of the public interface we need to implement to make a functional dag node.

Reply:
This is because upon executing the dag, we need to recursively find and replace nodes to corresponding _execute_impl output value. Then in https://sourcegraph.com/github.com/ray-project/ray/-/blob/python/ray/experimental/dag/dag_node.py?L174 we're returning a new instance with replaced args / kwargs.

Reply:
The first part of what you said makes sense, but the second part I don't quite understand. Shouldn't we only need to call `_execute_impl` in the execution path? Why do we need to "return a new instance with replaced args/kwargs"?